### PR TITLE
KM_variance_estimator

### PR DESCRIPTION
**Pull request by TristanFauvel**
**Checklist**

- [x] py.test passes
- [x] tests are included
- [x] code is well formatted
- [ ] documentation renders correctly

**What does this implement/fix? Explain your changes**

Hi @sebp,
I added the Greenwood's estimation of KM variance to nonparametric.py (this is a prerequesite for implementing some goodness-of-fit tests).
NB:  I ran tox -e py310-docs but for some reason the new function does not not appear in the API doc. 
Best,

### CODE REVIEW

**Review by sebp — CHANGES_REQUESTED** [review]
Thanks for your PR!

Could you please revert the code style changes and `prod.txt` so it is easier to see the relevant changes.

Moreover, some of your code is not covered by tests. I'd appreciate if you could provide tests that cover all your code.

**Review by sebp — CHANGES_REQUESTED** [review]
Thanks for updating the code.

Currently, there's a lot of duplicate code shared with `kaplan_meier_estimator`. Therefore, I would suggest to add the code for variance estimation to `kaplan_meier_estimator`.

File: sksurv/nonparametric.py [hunk 1]
```diff
@@ -254,6 +257,90 @@ def kaplan_meier_estimator(event, time_exit, time_enter=None, time_min=None, rev
...
+        out=np.zeros(uniq_times.shape[0], dtype=float),
+        where=n_events != 0,
+    )
+    values_rem = 1.0 - ratio_events
+
+    ratio_var = np.divide(
+        n_events,
+        n_at_risk * (n_at_risk - n_events),
+        out=np.zeros(uniq_times.shape[0], dtype=float),
+        where=(n_events != 0) & (n_at_risk != n_events),
```
[the remark this answers is not in the document]
Comment (TristanFauvel) [reply sksurv/nonparametric.py]: In TestKaplanMeier, test_simple for data_all_uncensored includes the scenario where `n_at_risk == n_events`.

**Comment by sebp** [comment]
@TristanFauvel Could you please rebase on latest master, which fixes a couple of issues with CI.

**Comment by sebp** [comment]
FYI, your changes look good. Thanks so much for contributing 🤩

Since variance isn't all that useful on its own, I'm planning to a add support for confidence intervals based on your code.

**Pull request closed without merging**